// portfolio/src/components/ContactForm.tsx
import React, { useState, useEffect } from 'react';
import { motion } from 'framer-motion';
import { MapPin, Phone, Mail, Clock, MessageSquare, PhoneCall, Send, ArrowRight } from 'lucide-react';
import { translations } from '../data/translations';

interface ContactFormProps {
  currentLang: 'en' | 'ar';
}

export const ContactForm: React.FC<ContactFormProps> = ({ currentLang: initialLang }) => {
  const [lang, setLang] = useState<'en' | 'ar'>(initialLang);
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    phone: '',
    company: '',
    city: '',
    country: 'Saudi Arabia',
    industry: '',
    product: '',
    message: '',
    contactMethod: 'whatsapp'
  });
  const [status, setStatus] = useState<'idle' | 'sending' | 'success' | 'error'>('idle');

  // Sync language selection
  useEffect(() => {
    const handleLangChange = (e: Event) => {
      const customEvent = e as CustomEvent<'en' | 'ar'>;
      setLang(customEvent.detail);
    };
    window.addEventListener('lang-change', handleLangChange);
    return () => window.removeEventListener('lang-change', handleLangChange);
  }, []);

  const t = translations[lang].contact;

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => {
    const { name, value } = e.target;
    setFormData((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (!formData.name || !formData.email || !formData.phone || !formData.message) {
      setStatus('error');
      return;
    }

    setStatus('sending');

    const emailTo = 'optimumsupply313@gmail.com';
    const whatsappPhone = '966563539227';
    let emailSubject = '';
    let bodyText = '';

    if (lang === 'ar') {
      emailSubject = `استفسار جديد من موقع أوبتيموم للتوريد - ${formData.company || formData.name}`;
      bodyText = `مرحباً فريق أوبتيموم للتوريد التجاري،\n\n` +
                  `تم إرسال نموذج تواصل واستفسار جديد عبر الموقع الإلكتروني:\n\n` +
                  `- الاسم: ${formData.name}\n` +
                  `- الشركة: ${formData.company || 'غير محدد'}\n` +
                  `- الجوال: ${formData.phone}\n` +
                  `- البريد الإلكتروني: ${formData.email}\n` +
                  `- الموقع: ${formData.city || 'غير محدد'}، ${formData.country}\n` +
                  `- القطاع: ${formData.industry || 'غير محدد'}\n` +
                  `- المنتج المستهدف: ${formData.product || 'غير محدد'}\n` +
                  `- وسيلة التواصل المفضلة: ${formData.contactMethod}\n\n` +
                  `تفاصيل الاستفسار:\n${formData.message}\n`;
    } else {
      emailSubject = `New Sourcing Inquiry - ${formData.company || formData.name}`;
      bodyText = `Hello Optimum Supply Trading Est. Team,\n\n` +
                  `You have received a new contact/sourcing inquiry from the website:\n\n` +
                  `- Name: ${formData.name}\n` +
                  `- Company: ${formData.company || 'Not provided'}\n` +
                  `- Phone: ${formData.phone}\n` +
                  `- Email: ${formData.email}\n` +
                  `- Location: ${formData.city || 'Not provided'}, ${formData.country}\n` +
                  `- Industry: ${formData.industry || 'Not provided'}\n` +
                  `- Product of Interest: ${formData.product || 'Not provided'}\n` +
                  `- Preferred Contact: ${formData.contactMethod}\n\n` +
                  `Message:\n${formData.message}\n`;
    }

    const mailtoUrl = `mailto:${emailTo}?subject=${encodeURIComponent(emailSubject)}&body=${encodeURIComponent(bodyText)}`;
    const whatsappUrl = `https://wa.me/${whatsappPhone}?text=${encodeURIComponent(bodyText)}`;
    
    // Redirect to user choice after a short timeout
    setTimeout(() => {
      if (formData.contactMethod === 'whatsapp') {
        window.open(whatsappUrl, '_blank');
      } else {
        window.location.href = mailtoUrl;
      }
      setStatus('success');
      setFormData({
        name: '',
        email: '',
        phone: '',
        company: '',
        city: '',
        country: 'Saudi Arabia',
        industry: '',
        product: '',
        message: '',
        contactMethod: 'whatsapp'
      });
      setTimeout(() => setStatus('idle'), 6000);
    }, 800);
  };

  return (
    <section id="contact" className="py-24 bg-canvas-soft border-t border-glass-border select-none">
      <div className="max-w-7xl mx-auto px-6">
        
        {/* Section Header */}
        <div className="flex flex-col items-center text-center mb-16">
          <span className="caption-mono text-accent-blue mb-3 block">
            {t.badge}
          </span>
          <h2 className="display-lg text-ink">
            {t.headline}
          </h2>
        </div>

        {/* Dual column layout split */}
        <div className="grid grid-cols-1 lg:grid-cols-12 gap-8 max-w-6xl mx-auto mb-12">
          
          {/* Left Column: Direct Contacts */}
          <div className="lg:col-span-5 flex flex-col gap-4">
            
            {/* Address — clicking opens exact location in Google Maps */}
            <a
              href="https://www.google.com/maps?q=26.4437545,50.1104749"
              target="_blank"
              rel="noopener noreferrer"
              className="p-5 glass-panel rounded-2xl flex gap-4 border-l-2 border-l-accent-blue cursor-pointer hover:ring-2 hover:ring-accent-blue/40 hover:border-l-accent-cyan transition-all duration-200 group/addr"
            >
              <div className="w-10 h-10 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border text-accent-blue flex-shrink-0 group-hover/addr:bg-accent-blue/20 transition-colors">
                <MapPin className="w-5 h-5 group-hover/addr:animate-bounce" />
              </div>
              <div>
                <h4 className="text-xs font-bold text-ink mb-1 flex items-center gap-1.5">
                  {t.addressTitle}
                  <span className="text-[9px] text-accent-cyan font-mono normal-case">↗ Open in Maps</span>
                </h4>
                <p className="text-[11px] text-body leading-relaxed">{t.addressText}</p>
                <div className="mt-3 flex flex-wrap gap-2">
                  <span className="px-2.5 py-1 rounded bg-gradient-to-r from-accent-blue to-accent-cyan text-white text-[9px] font-extrabold uppercase font-mono shadow-sm shadow-accent-blue/15">
                    GPS: 26.4437545, 50.1104749
                  </span>
                  <span className="px-2.5 py-1 rounded bg-gradient-to-r from-amber-500 to-orange-600 text-white text-[9px] font-extrabold uppercase font-mono shadow-sm shadow-orange-500/15">
                    C4V6+G55 Dammam Saudi Arabia
                  </span>
                </div>
              </div>
            </a>


            {/* Phone / Whatsapp */}
            <div className="p-5 glass-panel rounded-2xl flex gap-4 border-l-2 border-l-accent-cyan">
              <div className="w-10 h-10 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border text-accent-cyan flex-shrink-0">
                <Phone className="w-5 h-5" />
              </div>
              <div>
                <h4 className="text-xs font-bold text-ink mb-1">{t.phoneTitle}</h4>
                <p className="text-[11px] text-body leading-relaxed">{t.phoneText}</p>
              </div>
            </div>

            {/* Email Desk */}
            <div className="p-5 glass-panel rounded-2xl flex gap-4 border-l-2 border-l-purple-500">
              <div className="w-10 h-10 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border text-purple-500 flex-shrink-0">
                <Mail className="w-5 h-5" />
              </div>
              <div>
                <h4 className="text-xs font-bold text-ink mb-1">{t.emailTitle}</h4>
                <p className="text-[11px] text-body leading-relaxed">{t.emailText}</p>
              </div>
            </div>

            {/* Hours */}
            <div className="p-5 glass-panel rounded-2xl flex gap-4 border-l-2 border-l-amber-500">
              <div className="w-10 h-10 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border text-amber-500 flex-shrink-0">
                <Clock className="w-5 h-5" />
              </div>
              <div>
                <h4 className="text-xs font-bold text-ink mb-1">{t.hoursTitle}</h4>
                <p className="text-[11px] text-body leading-relaxed">{t.hoursText}</p>
              </div>
            </div>

            {/* Quick Actions Shortcuts */}
            <div className="grid grid-cols-3 gap-2 mt-2">
              <a
                href="https://wa.me/966563539227"
                target="_blank"
                rel="noopener noreferrer"
                className="flex flex-col items-center justify-center p-3 rounded-xl border border-glass-border hover:bg-glass-border/30 text-[10px] font-bold text-ink text-center gap-1.5 focus:outline-none"
              >
                <MessageSquare className="w-4 h-4 text-emerald-500" />
                <span>{t.whatsappBtn}</span>
              </a>
              <a
                href="mailto:optimumsupply313@gmail.com"
                className="flex flex-col items-center justify-center p-3 rounded-xl border border-glass-border hover:bg-glass-border/30 text-[10px] font-bold text-ink text-center gap-1.5 focus:outline-none"
              >
                <Mail className="w-4 h-4 text-accent-blue" />
                <span>{t.emailBtn}</span>
              </a>
              <a
                href="tel:+966563539227"
                className="flex flex-col items-center justify-center p-3 rounded-xl border border-glass-border hover:bg-glass-border/30 text-[10px] font-bold text-ink text-center gap-1.5 focus:outline-none"
              >
                <PhoneCall className="w-4 h-4 text-accent-cyan" />
                <span>{t.callBtn}</span>
              </a>
            </div>

          </div>

          {/* Right Column: Sourcing Inquiry Form */}
          <div className="lg:col-span-7">
            <div className="glass-panel p-6 rounded-2xl border-t border-t-accent-blue/30 h-full">
              <h3 className="text-sm font-bold text-ink mb-5 font-sans border-b border-glass-border pb-2">
                {t.formTitle}
              </h3>

              <form onSubmit={handleSubmit} className="space-y-4">
                {/* Name & Email */}
                <div className="grid grid-cols-1 sm:grid-cols-2 gap-4">
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{t.formName} *</label>
                    <input
                      type="text"
                      name="name"
                      value={formData.name}
                      onChange={handleInputChange}
                      required
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{t.formEmail} *</label>
                    <input
                      type="email"
                      name="email"
                      value={formData.email}
                      onChange={handleInputChange}
                      required
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue text-left"
                    />
                  </div>
                </div>

                {/* Phone & Company */}
                <div className="grid grid-cols-1 sm:grid-cols-2 gap-4">
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{t.formPhone} *</label>
                    <input
                      type="text"
                      name="phone"
                      value={formData.phone}
                      onChange={handleInputChange}
                      required
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue text-left"
                    />
                  </div>
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{t.formCompany}</label>
                    <input
                      type="text"
                      name="company"
                      value={formData.company}
                      onChange={handleInputChange}
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                </div>

                {/* City & Country */}
                <div className="grid grid-cols-1 sm:grid-cols-2 gap-4">
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{lang === 'ar' ? 'المدينة' : 'City'}</label>
                    <input
                      type="text"
                      name="city"
                      value={formData.city}
                      onChange={handleInputChange}
                      placeholder="e.g. Dammam, Riyadh"
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{lang === 'ar' ? 'الدولة' : 'Country'}</label>
                    <input
                      type="text"
                      name="country"
                      value={formData.country}
                      onChange={handleInputChange}
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                </div>

                {/* Industry & Product Interest */}
                <div className="grid grid-cols-1 sm:grid-cols-2 gap-4">
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{lang === 'ar' ? 'القطاع الصناعي' : 'Industry Segment'}</label>
                    <input
                      type="text"
                      name="industry"
                      value={formData.industry}
                      onChange={handleInputChange}
                      placeholder="e.g. Oil & Gas, Construction"
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                  <div className="flex flex-col">
                    <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{lang === 'ar' ? 'المنتج المهتم به' : 'Product of Interest'}</label>
                    <input
                      type="text"
                      name="product"
                      value={formData.product}
                      onChange={handleInputChange}
                      placeholder="e.g. Industrial Valves, PPE"
                      className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue"
                    />
                  </div>
                </div>

                {/* Message / Specifications */}
                <div className="flex flex-col">
                  <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{t.formMessage} *</label>
                  <textarea
                    name="message"
                    value={formData.message}
                    onChange={handleInputChange}
                    required
                    rows={4}
                    className="px-3.5 py-2 text-xs rounded-lg border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue resize-none"
                  />
                </div>

                {/* Preferred Contact Method */}
                <div className="flex flex-col">
                  <label className="text-[9px] font-mono text-mute uppercase mb-1 font-bold">{lang === 'ar' ? 'وسيلة التواصل المفضلة' : 'Preferred Contact Method'}</label>
                  <select
                    name="contactMethod"
                    value={formData.contactMethod}
                    onChange={handleInputChange}
                    className="px-3.5 py-2.5 text-xs rounded-lg border border-glass-border bg-canvas text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue font-semibold cursor-pointer"
                  >
                    <option value="whatsapp">{lang === 'ar' ? 'الواتساب (الأسرع)' : 'WhatsApp (Fastest)'}</option>
                    <option value="email">{lang === 'ar' ? 'البريد الإلكتروني' : 'Email'}</option>
                  </select>
                </div>

                {/* Submit button */}
                <button
                  type="submit"
                  disabled={status === 'sending' || status === 'success'}
                  className="w-full px-5 py-3 rounded-lg bg-gradient-to-r from-accent-blue to-accent-cyan text-white text-xs font-bold hover:opacity-95 disabled:opacity-50 transition-opacity flex items-center justify-center gap-1.5 focus:outline-none cursor-pointer"
                >
                  {status === 'sending' ? (
                    <span>{t.formSending}</span>
                  ) : (
                    <>
                      <span>{t.formSubmit}</span>
                      <ArrowRight className="w-4 h-4 rtl:rotate-180" />
                    </>
                  )}
                </button>

                {/* Status response info */}
                {status === 'success' && (
                  <p className="text-xs text-emerald-500 font-semibold text-center mt-3 font-mono">
                    {t.formSuccess}
                  </p>
                )}
                {status === 'error' && (
                  <p className="text-xs text-rose-500 font-semibold text-center mt-3 font-mono">
                    {t.formError}
                  </p>
                )}
              </form>
            </div>
          </div>

        </div>

        {/* Embedded Map Section */}
        <div className="max-w-6xl mx-auto glass-panel p-2 rounded-2xl overflow-hidden shadow-md">
          <iframe
            src="https://maps.google.com/maps?q=26.4437545,50.1104749&z=15&output=embed"
            className="w-full h-80 rounded-xl border-0 filter grayscale dark:invert-[90%] dark:hue-rotate-[180deg] select-none"
            allowFullScreen
            loading="lazy"
            referrerPolicy="no-referrer-when-downgrade"
            title="Optimum Supply Trading Est. Office Map"
          />
        </div>

      </div>
    </section>
  );
};
export default ContactForm;
