// portfolio/src/components/Industries.tsx
import React, { useState, useEffect } from 'react';
import { motion } from 'framer-motion';
import { 
  Flame, Atom, Wrench, Factory, Anchor, Zap, Waves, Briefcase, Landmark, ShieldCheck
} from 'lucide-react';
import { translations } from '../data/translations';

interface IndustriesProps {
  currentLang: 'en' | 'ar';
}

export const Industries: React.FC<IndustriesProps> = ({ currentLang: initialLang }) => {
  const [lang, setLang] = useState<'en' | 'ar'>(initialLang);

  // Sync language selection
  useEffect(() => {
    const handleLangChange = (e: Event) => {
      const customEvent = e as CustomEvent<'en' | 'ar'>;
      setLang(customEvent.detail);
    };
    window.addEventListener('lang-change', handleLangChange);
    return () => window.removeEventListener('lang-change', handleLangChange);
  }, []);

  const t = translations[lang].industries;

  // Custom mouse coordinates for card lighting glow
  const handleMouseMove = (e: React.MouseEvent<HTMLDivElement>) => {
    const card = e.currentTarget;
    const rect = card.getBoundingClientRect();
    const x = e.clientX - rect.left;
    const y = e.clientY - rect.top;
    card.style.setProperty('--mouse-x', `${x}px`);
    card.style.setProperty('--mouse-y', `${y}px`);
  };

  // 10 sectors list with icons and border colors
  const sectorsList = [
    { key: 'oilGas', icon: <Flame className="w-5 h-5" />, color: 'border-l-red-500', text: t.oilGas },
    { key: 'petro', icon: <Atom className="w-5 h-5" />, color: 'border-l-purple-500', text: t.petro },
    { key: 'const', icon: <Wrench className="w-5 h-5" />, color: 'border-l-amber-500', text: t.const },
    { key: 'mfg', icon: <Factory className="w-5 h-5" />, color: 'border-l-blue-500', text: t.mfg },
    { key: 'marine', icon: <Anchor className="w-5 h-5" />, color: 'border-l-cyan-500', text: t.marine },
    { key: 'power', icon: <Zap className="w-5 h-5" />, color: 'border-l-yellow-500', text: t.power },
    { key: 'water', icon: <Waves className="w-5 h-5" />, color: 'border-l-teal-500', text: t.water },
    { key: 'maint', icon: <ShieldCheck className="w-5 h-5" />, color: 'border-l-slate-400', text: t.maint },
    { key: 'epc', icon: <Briefcase className="w-5 h-5" />, color: 'border-l-indigo-500', text: t.epc },
    { key: 'govt', icon: <Landmark className="w-5 h-5" />, color: 'border-l-pink-500', text: t.govt }
  ];

  return (
    <section id="industries" className="py-24 bg-canvas border-t border-glass-border select-none overflow-hidden">
      <div className="max-w-7xl mx-auto px-6">
        
        {/* Section Header */}
        <div className="flex flex-col items-center text-center mb-16">
          <span className="caption-mono text-accent-blue mb-3 block">
            {t.badge}
          </span>
          <h2 className="display-lg text-ink">
            {t.headline}
          </h2>
        </div>

        {/* 10 industries grid layout */}
        <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4 max-w-6xl mx-auto">
          {sectorsList.map((sector, idx) => (
            <motion.div
              key={sector.key}
              initial={{ opacity: 0, y: 25 }}
              whileInView={{ opacity: 1, y: 0 }}
              viewport={{ once: true, margin: '-40px' }}
              transition={{ duration: 0.5, delay: idx * 0.02 }}
              onMouseMove={handleMouseMove}
              className={`glow-card p-5 glass-panel rounded-xl flex flex-col justify-between h-[210px] group cursor-default border-l-2 ${sector.color} relative overflow-hidden`}
            >
              {/* Engineering blueprint background wireframe */}
              <div className="absolute inset-0 bg-[linear-gradient(to_right,#ffffff01_1px,transparent_1px),linear-gradient(to_bottom,#ffffff01_1px,transparent_1px)] bg-[size:14px_14px] opacity-15 pointer-events-none z-0" />

              <div className="relative z-10">
                {/* Header card */}
                <div className="flex justify-between items-start mb-4">
                  <div className="w-9 h-9 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border shadow-sm text-ink group-hover:scale-105 group-hover:bg-glass-border/70 transition-all duration-300">
                    {sector.icon}
                  </div>
                  <span className="text-[9px] font-mono text-mute uppercase tracking-widest">
                    IND-0{idx + 1}
                  </span>
                </div>

                {/* Content info */}
                <h3 className="text-xs font-bold text-ink mb-1.5 group-hover:text-accent-cyan transition-colors">
                  {sector.text.title}
                </h3>
                <p className="text-[10px] text-body leading-relaxed line-clamp-3">
                  {sector.text.desc}
                </p>
              </div>

              {/* Glowing lower border lines indicator */}
              <div className="absolute bottom-0 left-0 right-0 h-[2px] bg-transparent overflow-hidden rounded-b-xl z-20">
                <div className="w-full h-full bg-gradient-to-r from-transparent via-accent-cyan to-transparent translate-x-[-100%] group-hover:translate-x-[100%] transition-transform duration-1000 ease-out" />
              </div>
            </motion.div>
          ))}
        </div>

      </div>
    </section>
  );
};
export default Industries;
