// portfolio/src/components/Navbar.tsx
import React, { useState, useEffect } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { Sun, Moon, Globe, Search, Menu, X, ChevronDown, Zap, Mail, MessageSquare } from 'lucide-react';
import { translations } from '../data/translations';

interface NavbarProps {
  currentLang: 'en' | 'ar';
  currentTheme: 'dark' | 'light';
  onRequestQuote: () => void;
}

export const Navbar: React.FC<NavbarProps> = ({ 
  currentLang: initialLang, 
  currentTheme: initialTheme,
  onRequestQuote
}) => {
  const [lang, setLang] = useState<'en' | 'ar'>(initialLang);
  const [theme, setTheme] = useState<'dark' | 'light'>(initialTheme);
  const [isScrolled, setIsScrolled] = useState(false);
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
  const [activeMegaMenu, setActiveMegaMenu] = useState<boolean>(false);
  const [searchOpen, setSearchOpen] = useState(false);
  const [searchQuery, setSearchQuery] = useState('');

  const t = translations[lang].nav;

  // Listen to scrolls to adjust navbar blur/height
  useEffect(() => {
    const handleScroll = () => {
      setIsScrolled(window.scrollY > 20);
    };
    window.addEventListener('scroll', handleScroll);
    return () => window.removeEventListener('scroll', handleScroll);
  }, []);

  // Update theme in DOM and dispatch custom event
  const toggleTheme = () => {
    const nextTheme = theme === 'dark' ? 'light' : 'dark';
    setTheme(nextTheme);
    localStorage.setItem('theme', nextTheme);
    
    if (nextTheme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    
    window.dispatchEvent(new CustomEvent('theme-change', { detail: nextTheme }));
  };

  // Update language, document direction, and dispatch custom event
  const toggleLang = () => {
    const nextLang = lang === 'en' ? 'ar' : 'en';
    setLang(nextLang);
    localStorage.setItem('lang', nextLang);
    document.documentElement.setAttribute('lang', nextLang);
    document.documentElement.setAttribute('dir', nextLang === 'ar' ? 'rtl' : 'ltr');
    
    window.dispatchEvent(new CustomEvent('lang-change', { detail: nextLang }));
  };

  // Close mega menu when clicking outside
  useEffect(() => {
    const handleOutsideClick = () => {
      setActiveMegaMenu(false);
    };
    document.addEventListener('click', handleOutsideClick);
    return () => document.removeEventListener('click', handleOutsideClick);
  }, []);

  const handleSearchSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (searchQuery.trim()) {
      window.dispatchEvent(new CustomEvent('search-submit', { detail: searchQuery }));
      const productsSection = document.getElementById('products');
      if (productsSection) {
        productsSection.scrollIntoView({ behavior: 'smooth' });
      }
    }
  };

  return (
    <header 
      className={`fixed top-0 left-0 right-0 z-40 transition-all duration-300 ${
        isScrolled 
          ? 'py-3 bg-glass-bg backdrop-blur-md border-b border-glass-border shadow-md' 
          : 'py-5 bg-transparent'
      }`}
    >
      <div className="max-w-7xl mx-auto px-6 flex items-center justify-between">
        {/* Brand Logo */}
        <a 
          href="#" 
          className="flex items-center gap-2 group focus:outline-none"
          onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })}
        >
          <div className="relative w-14 h-14 rounded-xl bg-gradient-to-tr from-accent-blue to-accent-cyan flex items-center justify-center shadow-lg shadow-accent-cyan/20 group-hover:scale-105 transition-transform overflow-hidden">
            <img src="/globe_logo.png" className="w-full h-full object-contain" alt="Optimum Supply Logo" />
          </div>
          <span className="font-sans font-extrabold text-base md:text-lg tracking-tight select-none">
            {lang === 'ar' ? (
              <span className="text-ink">أوبتيموم <span className="text-accent-cyan font-bold">للتوريد</span></span>
            ) : (
              <span className="text-ink">OPTIMUM <span className="text-accent-blue font-bold">SUPPLY</span></span>
            )}
          </span>
        </a>

        {/* Desktop Nav Navigation links */}
        <nav className="hidden lg:flex items-center gap-1">
          <a href="#" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.home}
          </a>
          <a href="#about" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.about}
          </a>
          
          {/* Mega Menu Toggle Link */}
          <div className="relative" onClick={(e) => e.stopPropagation()}>
            <button 
              onClick={() => setActiveMegaMenu(!activeMegaMenu)}
              className={`flex items-center gap-1 px-3 py-1.5 text-xs font-semibold rounded-full transition-colors focus:outline-none ${
                activeMegaMenu 
                  ? 'text-ink bg-glass-border/40' 
                  : 'text-body hover:text-ink hover:bg-glass-border/30'
              }`}
            >
              {t.products}
              <ChevronDown className={`w-3.5 h-3.5 transition-transform duration-200 ${activeMegaMenu ? 'rotate-180' : ''}`} />
            </button>

            {/* Mega Menu Dropdown */}
            <AnimatePresence>
              {activeMegaMenu && (
                <motion.div
                  initial={{ opacity: 0, y: 15 }}
                  animate={{ opacity: 1, y: 0 }}
                  exit={{ opacity: 0, y: 10 }}
                  transition={{ duration: 0.2 }}
                  className="absolute left-1/2 -translate-x-1/2 top-12 w-[600px] p-6 glass-panel rounded-xl grid grid-cols-2 gap-6 z-50 text-ink"
                >
                  {/* Left Column Categories */}
                  <div>
                    <h3 className="text-xs font-bold uppercase tracking-wider text-accent-blue font-mono border-b border-glass-border pb-1 mb-3">
                      {lang === 'ar' ? 'أقسام التوريد' : 'Supply Streams'}
                    </h3>
                    <ul className="space-y-2">
                      {[
                        { key: 'industrial', name: lang === 'ar' ? 'التوريدات الصناعية' : 'Industrial Supplies' },
                        { key: 'mechanical', name: lang === 'ar' ? 'المنتجات الميكانيكية' : 'Mechanical Products' },
                        { key: 'electrical', name: lang === 'ar' ? 'المعدات الكهربائية' : 'Electrical Supplies' },
                        { key: 'safety', name: lang === 'ar' ? 'معدات السلامة' : 'Safety Equipment' }
                      ].map((item) => (
                        <li key={item.key}>
                          <a 
                            href={`#products`} 
                            onClick={() => setActiveMegaMenu(false)}
                            className="text-xs text-body hover:text-ink hover:underline block py-1"
                          >
                            {item.name}
                          </a>
                        </li>
                      ))}
                    </ul>
                  </div>

                  {/* Right Column Categories */}
                  <div>
                    <h3 className="text-xs font-bold uppercase tracking-wider text-accent-cyan font-mono border-b border-glass-border pb-1 mb-3">
                      {lang === 'ar' ? 'المواد العامة والإنشائية' : 'General & Materials'}
                    </h3>
                    <ul className="space-y-2">
                      {[
                        { key: 'welding', name: lang === 'ar' ? 'معدات اللحام' : 'Welding Equipment' },
                        { key: 'building', name: lang === 'ar' ? 'مواد البناء' : 'Building Materials' },
                        { key: 'general', name: lang === 'ar' ? 'تجارة عامة' : 'General Trading' }
                      ].map((item) => (
                        <li key={item.key}>
                          <a 
                            href={`#products`} 
                            onClick={() => setActiveMegaMenu(false)}
                            className="text-xs text-body hover:text-ink hover:underline block py-1"
                          >
                            {item.name}
                          </a>
                        </li>
                      ))}
                    </ul>
                  </div>
                </motion.div>
              )}
            </AnimatePresence>
          </div>

          <a href="#industries" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.industries}
          </a>
          <a href="#projects" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.projects}
          </a>
          <a href="#why-us" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.whyUs}
          </a>
          <a href="#contact" className="px-3 py-1.5 text-xs font-semibold text-body hover:text-ink rounded-full hover:bg-glass-border/30 transition-colors">
            {t.contact}
          </a>
        </nav>

        {/* Right side controls: Search, Action shortcuts, Lang, Theme */}
        <div className="flex items-center gap-2">
          
          {/* Action CTAs (Desktop Only) */}
          <div className="hidden xl:flex items-center gap-2 mr-2">
            {/* WhatsApp */}
            <a
              href="https://wa.me/966563539227"
              target="_blank"
              rel="noopener noreferrer"
              className="px-3 py-1.5 rounded-full bg-gradient-to-r from-emerald-500 to-green-600 hover:opacity-95 text-white text-[10px] font-extrabold uppercase tracking-wider flex items-center gap-1 shadow-sm transition-all hover:scale-105 active:scale-95 cursor-pointer"
            >
              <MessageSquare className="w-3.5 h-3.5" />
              <span>{t.whatsapp}</span>
            </a>
            {/* Email */}
            <a
              href="mailto:optimumsupply313@gmail.com"
              className="px-3 py-1.5 rounded-full border border-glass-border bg-canvas/30 hover:bg-glass-border/50 text-ink text-[10px] font-extrabold uppercase tracking-wider flex items-center gap-1 transition-all hover:scale-105 active:scale-95"
            >
              <Mail className="w-3.5 h-3.5 text-accent-blue" />
              <span>{t.emailUs}</span>
            </a>
            {/* Quote modal trigger */}
            <button
              onClick={onRequestQuote}
              className="px-3 py-1.5 rounded-full bg-gradient-to-r from-accent-blue to-accent-cyan text-white text-[10px] font-extrabold uppercase tracking-wider flex items-center gap-1 shadow-sm transition-all hover:scale-105 active:scale-95 cursor-pointer"
            >
              <Zap className="w-3.5 h-3.5 text-amber-300" />
              <span>{t.requestQuote}</span>
            </button>
          </div>

          {/* Search Trigger */}
          <form onSubmit={handleSearchSubmit} className="relative flex items-center">
            <AnimatePresence>
              {searchOpen && (
                <motion.input
                  initial={{ width: 0, opacity: 0 }}
                  animate={{ width: 140, opacity: 1 }}
                  exit={{ width: 0, opacity: 0 }}
                  type="text"
                  placeholder={t.searchPlaceholder}
                  value={searchQuery}
                  onChange={(e) => setSearchQuery(e.target.value)}
                  className="px-3 py-1.5 text-xs rounded-full border border-glass-border bg-glass-bg text-ink focus:outline-none focus:ring-1 focus:ring-accent-blue mr-2 text-left rtl:text-right"
                  autoFocus
                />
              )}
            </AnimatePresence>
            <button 
              type="button"
              onClick={() => {
                if (searchOpen && searchQuery) {
                  window.dispatchEvent(new CustomEvent('search-submit', { detail: searchQuery }));
                  const prod = document.getElementById('products');
                  if (prod) prod.scrollIntoView({ behavior: 'smooth' });
                } else {
                  setSearchOpen(!searchOpen);
                }
              }}
              className="p-2 rounded-full hover:bg-glass-border/30 text-body hover:text-ink transition-colors focus:outline-none"
              aria-label="Search products"
            >
              <Search className="w-4.5 h-4.5" />
            </button>
          </form>

          {/* Language Swap Switcher */}
          <button
            onClick={toggleLang}
            className="p-2 rounded-full hover:bg-glass-border/30 text-body hover:text-ink transition-colors focus:outline-none flex items-center gap-1 text-xs font-mono font-bold"
            title="Switch Language"
            aria-label="Switch Language"
          >
            <Globe className="w-4.5 h-4.5" />
            <span className="uppercase">{lang === 'en' ? 'AR' : 'EN'}</span>
          </button>

          {/* Theme Toggler */}
          <button
            onClick={toggleTheme}
            className="p-2 rounded-full hover:bg-glass-border/30 text-body hover:text-ink transition-colors focus:outline-none"
            aria-label="Toggle Theme"
          >
            {theme === 'dark' ? <Sun className="w-4.5 h-4.5 text-amber-400" /> : <Moon className="w-4.5 h-4.5" />}
          </button>

          {/* Hamburger (Mobile Menu Toggler) */}
          <button
            onClick={() => setIsMobileMenuOpen(!isMobileMenuOpen)}
            className="p-2 rounded-full hover:bg-glass-border/30 text-body hover:text-ink transition-colors focus:outline-none lg:hidden"
            aria-label="Toggle Menu"
          >
            {isMobileMenuOpen ? <X className="w-5 h-5" /> : <Menu className="w-5 h-5" />}
          </button>
        </div>
      </div>

      {/* Mobile Drawer Menu */}
      <AnimatePresence>
        {isMobileMenuOpen && (
          <motion.div
            initial={{ opacity: 0, height: 0 }}
            animate={{ opacity: 1, height: 'auto' }}
            exit={{ opacity: 0, height: 0 }}
            className="lg:hidden border-t border-glass-border bg-glass-bg backdrop-blur-lg overflow-hidden text-ink"
          >
            <nav className="flex flex-col px-6 py-4 space-y-3 font-semibold">
              {[
                { href: '#', label: t.home },
                { href: '#about', label: t.about },
                { href: '#products', label: t.products },
                { href: '#industries', label: t.industries },
                { href: '#projects', label: t.projects },
                { href: '#why-us', label: t.whyUs },
                { href: '#contact', label: t.contact }
              ].map((item) => (
                <a
                  key={item.href}
                  href={item.href}
                  onClick={() => setIsMobileMenuOpen(false)}
                  className="py-1.5 text-sm hover:text-accent-cyan transition-colors"
                >
                  {item.label}
                </a>
              ))}
              
              {/* Mobile CTA shortcuts */}
              <div className="flex flex-col gap-2 pt-3 border-t border-glass-border">
                <button
                  onClick={() => {
                    setIsMobileMenuOpen(false);
                    onRequestQuote();
                  }}
                  className="w-full py-2.5 rounded-xl bg-gradient-to-r from-accent-blue to-accent-cyan text-white text-xs font-bold text-center cursor-pointer"
                >
                  {t.requestQuote}
                </button>
                <a
                  href="https://wa.me/966563539227"
                  target="_blank"
                  rel="noopener noreferrer"
                  className="w-full py-2.5 rounded-xl bg-green-600 text-white text-xs font-bold text-center flex items-center justify-center gap-1"
                >
                  <MessageSquare className="w-4 h-4" />
                  <span>{t.whatsapp}</span>
                </a>
              </div>
            </nav>
          </motion.div>
        )}
      </AnimatePresence>
    </header>
  );
};
export default Navbar;
