// portfolio/src/components/ProductShowcase.tsx
import React, { useState, useEffect } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { 
  Lightbulb, Cable, Layers, Grid, Zap, Cpu, 
  ShieldAlert, Box, Wrench, Container, Droplet, Eye, X, Send, Mail,
  Settings, Sliders, Wind, Flame, Thermometer, Shield
} from 'lucide-react';
import { translations } from '../data/translations';

interface ProductShowcaseProps {
  currentLang: 'en' | 'ar';
  onRequestQuote: () => void;
}

interface ProductItem {
  id: string;
  category: 'industrial' | 'mechanical' | 'electrical' | 'safety' | 'welding' | 'building' | 'general';
  name: string;
  desc: string;
  icon: React.ReactNode;
  specs: string[];
  applications: string[];
}

export const ProductShowcase: React.FC<ProductShowcaseProps> = ({ 
  currentLang: initialLang,
  onRequestQuote
}) => {
  const [lang, setLang] = useState<'en' | 'ar'>(initialLang);
  const [filter, setFilter] = useState<'all' | 'industrial' | 'mechanical' | 'electrical' | 'safety' | 'welding' | 'building' | 'general'>('all');
  const [searchQuery, setSearchQuery] = useState('');
  const [selectedProduct, setSelectedProduct] = useState<ProductItem | null>(null);

  // Sync language selection
  useEffect(() => {
    const handleLangChange = (e: Event) => {
      const customEvent = e as CustomEvent<'en' | 'ar'>;
      setLang(customEvent.detail);
    };
    window.addEventListener('lang-change', handleLangChange);
    return () => window.removeEventListener('lang-change', handleLangChange);
  }, []);

  // Listen to search events dispatched from the Navbar
  useEffect(() => {
    const handleSearch = (e: Event) => {
      const customEvent = e as CustomEvent<string>;
      setSearchQuery(customEvent.detail);
    };
    window.addEventListener('search-submit', handleSearch);
    return () => window.removeEventListener('search-submit', handleSearch);
  }, []);

  const t = translations[lang].products;

  // Custom Category Names mappings for filters
  const categoryLabels = {
    en: {
      industrial: "Industrial Supplies",
      mechanical: "Mechanical Products",
      electrical: "Electrical Supplies",
      safety: "Safety Equipment",
      welding: "Welding Equipment",
      building: "Building Materials",
      general: "General Trading"
    },
    ar: {
      industrial: "التوريدات الصناعية",
      mechanical: "المنتجات الميكانيكية",
      electrical: "المعدات الكهربائية",
      safety: "معدات السلامة",
      welding: "معدات اللحام",
      building: "مواد البناء",
      general: "تجارة عامة"
    }
  };

  const cats = categoryLabels[lang];

  // Icons map matching translations key
  const iconsMap: Record<string, React.ReactNode> = {
    valves: <Sliders className="w-6 h-6 text-accent-blue" />,
    pneumatic: <Settings className="w-6 h-6 text-accent-cyan" />,
    pipes: <Droplet className="w-6 h-6 text-accent-blue" />,
    flanges: <Layers className="w-6 h-6 text-slate-400" />,
    cables: <Cable className="w-6 h-6 text-accent-blue" />,
    switchgear: <Cpu className="w-6 h-6 text-purple-500" />,
    ppe: <Shield className="w-6 h-6 text-emerald-500" />,
    gasDetect: <Wind className="w-6 h-6 text-amber-500" />,
    weldingMachines: <Zap className="w-6 h-6 text-orange-500" />,
    electrodes: <Flame className="w-6 h-6 text-red-500" />,
    buildingMats: <Grid className="w-6 h-6 text-slate-400" />,
    insulation: <Thermometer className="w-6 h-6 text-blue-400" />,
    procure: <Wrench className="w-6 h-6 text-indigo-400" />,
    consolidated: <Container className="w-6 h-6 text-amber-500" />
  };

  // 14 High-fidelity industrial product catalog items
  const productsList: ProductItem[] = [
    {
      id: 'valves',
      category: 'industrial',
      name: lang === 'ar' ? 'صمامات صناعية وأنظمة التحكم بالسوائل' : 'Industrial Valves & Fluid Control',
      desc: lang === 'ar' ? 'صمامات بوابية، كروية، وصمامات فراشة وفحص مصنعة من الفولاذ المقاوم للصدأ والبرونز لخطوط الضغط العالي بمصافي التكرير.' : 'Premium gate, ball, butterfly, and check valves constructed from cast steel, stainless steel, and bronze for high-pressure industrial lines.',
      icon: iconsMap.valves,
      specs: lang === 'ar'
        ? ['تصنيفات ضغط تصل إلى Class 2500 / PN400', 'أحجام تبدأ من 1/2 بوصة إلى 48 بوصة', 'معايير تصنيع API 6D و ASME B16.34']
        : ['Pressure ratings up to Class 2500 / PN400', 'Sizes ranging from 1/2" up to 48"', 'Compliance with API 6D and ASME B16.34 standards'],
      applications: lang === 'ar'
        ? ['مصافي النفط والغاز', 'خطوط النقل الكيميائية والبتروكيماوية', 'محطات معالجة وتحلية المياه']
        : ['Oil & Gas refineries', 'Chemical and petrochemical pipelines', 'Water treatment and desalination plants']
    },
    {
      id: 'pneumatic',
      category: 'industrial',
      name: lang === 'ar' ? 'المعدات الهيدروليكية والنيوماتيكية' : 'Hydraulic & Pneumatic Equipment',
      desc: lang === 'ar' ? 'مضخات هيدروليكية، اسطوانات، مشغلات هوائية وصمامات اتجاهية للآلات والمعدات التشغيلية.' : 'High-performance hydraulic pumps, cylinders, pneumatic actuators, and directional valves for machinery and factory lines.',
      icon: iconsMap.pneumatic,
      specs: lang === 'ar'
        ? ['ضغط تشغيل هيدروليكي يصل إلى 700 بار', 'أنابيب اسطوانات مصقولة وكروم صلب', 'صمامات نيوماتيكية موفرة للطاقة']
        : ['Hydraulic operating pressure up to 700 bar', 'Honed cylinder tubes and hard chrome rods', 'Energy-efficient pneumatic control valves'],
      applications: lang === 'ar'
        ? ['خطوط الإنتاج والتصنيع', 'الرافعات والمعدات الثقيلة', 'أنظمة التحكم بالبوابات الكبيرة']
        : ['Factory production lines', 'Heavy cranes and machinery', 'Large gate and actuator control systems']
    },
    {
      id: 'pipes',
      category: 'mechanical',
      name: lang === 'ar' ? 'أنابيب الفولاذ الكربوني وغير القابل للصدأ' : 'Carbon & Stainless Steel Pipes',
      desc: lang === 'ar' ? 'أنابيب فولاذ كربوني سلسة عالية التحمل وأنابيب غير قابلة للصدأ SS316 مقاومة للتآكل والصدأ لخطوط نقل المواد والغاز.' : 'Heavy-wall seamless carbon steel and corrosion-resistant stainless steel pipes for high-temperature and structural applications.',
      icon: iconsMap.pipes,
      specs: lang === 'ar'
        ? ['أنابيب غير ملحومة (Seamless) وبجداول مختلفة', 'درجات ASTM A106 Gr. B و ASTM A312 TP316L', 'شهادات اختبار المصنع الكاملة EN 10204 3.1']
        : ['Seamless configurations in various schedules', 'Grades ASTM A106 Gr. B and ASTM A312 TP316L', 'Full EN 10204 3.1 mill test certifications'],
      applications: lang === 'ar'
        ? ['شبكات توزيع السوائل والغاز', 'التطبيقات الهيكلية والإنشائية', 'خطوط البخار والضغط المرتفع']
        : ['Fluid and gas distribution lines', 'Structural and architectural applications', 'High-pressure steam systems']
    },
    {
      id: 'flanges',
      category: 'mechanical',
      name: lang === 'ar' ? 'الفلانجات والوصلات الصناعية' : 'Industrial Flanges & Gaskets',
      desc: lang === 'ar' ? 'فلانجات رقبة اللحام، فلانجات مسطحة وعمياء مع حشوات منع التسرب الحلزونية عالية الأداء.' : 'Weld neck, slip-on, and blind flanges paired with high-performance spiral wound gaskets for secure line terminations.',
      icon: iconsMap.flanges,
      specs: lang === 'ar'
        ? ['فلانجات معتمدة بمقاييس ASME B16.5', 'حشوات منع تسرب حلزونية SS316 مع حشوة جرافيت', 'طلاء مجلفن ومقاوم للصدأ']
        : ['Certified ASME B16.5 flange profiles', 'Spiral wound SS316 gaskets with graphite filler', 'Galvanized and rust-preventive coatings'],
      applications: lang === 'ar'
        ? ['توصيل الأنابيب والمحابس', 'منشآت معالجة المواد الكيماوية', 'مرافق تكرير وتسييل الغاز']
        : ['Pipe-to-valve connection joints', 'Chemical processing utilities', 'Gas liquefaction and refining units']
    },
    {
      id: 'cables',
      category: 'electrical',
      name: lang === 'ar' ? 'كابلات الجهد المنخفض والمتوسط' : 'Low & Medium Voltage Cables',
      desc: lang === 'ar' ? 'كابلات نقل الطاقة والتحكم للجهد المنخفض والمتوسط المعتمدة محلياً وعالمياً للعمل في أقسى بيئات التشغيل الصناعية.' : 'Low-voltage, medium-voltage power, and control cables certified to global standards for extreme industrial worksites.',
      icon: iconsMap.cables,
      specs: lang === 'ar'
        ? ['عزل XLPE ونواة نحاس أو ألومنيوم مسلحة (SWA)', 'فولتيات تشغيل تصل إلى 36 كيلو فولت', 'معتمدة من شركة الكهرباء السعودية SEC']
        : ['XLPE insulation with armored copper/aluminum (SWA)', 'Operating voltages up to 36 kV', 'SEC (Saudi Electricity Co.) approved and certified'],
      applications: lang === 'ar'
        ? ['محطات توزيع الكهرباء الفرعية', 'المصانع والمناجم الكبرى', 'تمديدات الكابلات الأرضية ومسارات الحوامل']
        : ['Electrical substations', 'Heavy factories and mining facilities', 'Underground power lines and tray systems']
    },
    {
      id: 'switchgear',
      category: 'electrical',
      name: lang === 'ar' ? 'لوحات التوزيع والمفاتيح الكهربائية' : 'Switchgear & Panel Boards',
      desc: lang === 'ar' ? 'لوحات تحكم وتوزيع طاقة كهربائية مخصصة مصنعة لحماية وموازنة الأحمال الكهربائية الكبرى.' : 'Custom-configured low-voltage distribution boards and switchgears engineered for safe power allocation and overload shielding.',
      icon: iconsMap.switchgear,
      specs: lang === 'ar'
        ? ['حماية قواطع مصبوبة MCCBs وقواطع هوائية ACBs', 'تصاميم مطابقة لمعايير IEC 61439', 'هياكل حماية IP65 معدنية أو بوليستر']
        : ['Molded Case and Air Circuit Breaker protection', 'Full compliance with IEC 61439 specifications', 'IP65 metal or polyester enclosures'],
      applications: lang === 'ar'
        ? ['مجمعات التوزيع الإدارية والصناعية', 'غرف التحكم والمولدات الاحتياطية', 'أنظمة المرافق والخدمات العامة']
        : ['Industrial power rooms', 'Generator backup control boards', 'Municipal and utility connections']
    },
    {
      id: 'ppe',
      category: 'safety',
      name: lang === 'ar' ? 'معدات الحماية الشخصية (PPE)' : 'Personal Protective Equipment (PPE)',
      desc: lang === 'ar' ? 'خوذات سلامة، نظارات واقية، سترات عاكسة، أحذية أمان فولاذية، وقفازات سلامة ومقاومة للحرارة لتأمين الكوادر في الميدان.' : 'Certified hard hats, safety glasses, high-visibility vests, chemical suits, steel-toe boots, and premium safety gloves for maximum site protection.',
      icon: iconsMap.ppe,
      specs: lang === 'ar'
        ? ['أحذية أمان بتصنيف حماية S3 ومقدمة صلبة', 'خوذات سلامة مقاومة للصدمات متوافقة مع ANSI Z89.1', 'سترات عاكسة عالية الجودة ومقاومة للاشتعال']
        : ['S3 rating safety shoes with steel toes', 'Impact-resistant hard hats complying with ANSI Z89.1', 'Flame-retardant high-visibility safety apparel'],
      applications: lang === 'ar'
        ? ['مواقع الإنشاءات والمقاولات', 'منشآت تصنيع النفط والمواد البترولية', 'المخازن ومناطق الشحن والتفريغ']
        : ['Construction and contracting sites', 'Petrochemical and drilling facilities', 'Warehousing and heavy material handling zones']
    },
    {
      id: 'gasDetect',
      category: 'safety',
      name: lang === 'ar' ? 'أنظمة الكشف عن الغازات والمخاطر' : 'Gas Detection Systems',
      desc: lang === 'ar' ? 'أجهزة كشف غازات محمولة وثابتة للكشف عن الغازات السامة والقابلة للاشتعال وحماية بيئة العمل.' : 'Portable and fixed-gas detection devices engineered to monitor combustible and toxic gases in closed industrial environments.',
      icon: iconsMap.gasDetect,
      specs: lang === 'ar'
        ? ['كشف متعدد للغازات (H2S, CO, O2, LEL)', 'تصنيف أمان عالي ATEX / IECEx للمناطق الخطرة', 'إنذارات صوتية وضوئية بقوة 95 ديسيبل']
        : ['Multi-gas sensing arrays (H2S, CO, O2, LEL)', 'ATEX / IECEx intrinsic safety ratings', '95 dB acoustic alarm with LED strobe alerts'],
      applications: lang === 'ar'
        ? ['الآبار والأنفاق المغلقة', 'خطوط الإنتاج والتوزيع الكيماوي', 'مصانع تكرير البترول والغاز الطبيعي']
        : ['Confined spaces and tunnels', 'Chemical process pipelines', 'Petroleum and natural gas refining plants']
    },
    {
      id: 'weldingMachines',
      category: 'welding',
      name: lang === 'ar' ? 'ماكينات اللحام ومستهلكاتها (MIG/TIG/ARC)' : 'Arc, TIG & MIG Welding Machines',
      desc: lang === 'ar' ? 'ماكينات لحام كهربائية مخصصة للأعمال الشاقة، وقواطع البلازما، وأسلاك وإلكترودات اللحام المعتمدة للمنشآت الحديدية الكبرى.' : 'Heavy-duty welding machines, plasma cutters, electrodes, and wires certified for high-strength steel welding and fabrication.',
      icon: iconsMap.weldingMachines,
      specs: lang === 'ar'
        ? ['تقنية العاكس الذكية (Inverter) لتوفير الطاقة', 'نطاق تيار لحام يبدأ من 30A إلى 600A', 'نظام تبريد قسري مدمج للحماية من الحرارة الزائدة']
        : ['Advanced energy-saving inverter technology', 'Amperage ranges from 30A up to 600A', 'Thermal overload protection with forced-air cooling'],
      applications: lang === 'ar'
        ? ['ورش الهياكل والإنشاءات المعدنية', 'بناء وصيانة الخزانات وخطوط الأنابيب', 'ورش التصنيع والمسابك الكبرى']
        : ['Steel structural fabrication shops', 'Tank and pipeline welding projects', 'Heavy manufacturing and foundry setups']
    },
    {
      id: 'electrodes',
      category: 'welding',
      name: lang === 'ar' ? 'أسلاك وإلكترودات اللحام' : 'Welding Electrodes & Wires',
      desc: lang === 'ar' ? 'أسلاك لحام MIG بكر وإلكترودات لحام مغطاة بمواصفات مطابقة لمقاييس AWS لقوة شد وجودة عالية.' : 'AWS-specification coated welding electrodes and spool wires for high-tensile structural and pipeline welding.',
      icon: iconsMap.electrodes,
      specs: lang === 'ar'
        ? ['إلكترودات E6013 و E7018 منخفضة الهيدروجين', 'سلك لحام MIG ER70S-6 عالي النقاوة', 'مقاومة عالية للتشقق والعيوب المسامية']
        : ['AWS E6013 and low-hydrogen E7018 electrodes', 'Premium spool MIG wire ER70S-6 carbon steel', 'Crack-resistant and low porosity weld deposits'],
      applications: lang === 'ar'
        ? ['لحام الفولاذ الكربوني والهياكل', 'توصيل وتثبيت أنابيب النفط والمياه', 'أعمال الصيانة المعدنية العامة']
        : ['Carbon steel and structural welding', 'High-pressure pipeline joining', 'General maintenance fabrication']
    },
    {
      id: 'buildingMats',
      category: 'building',
      name: lang === 'ar' ? 'الحديد الهيكلي ومواد التشييد' : 'Structural Steel & Construction Materials',
      desc: lang === 'ar' ? 'حديد تسليح عالي التحمل، جسور حديدية، ومواد إحكام كيميائية، ومواد عزل مقاومة للحرائق والحرارة العالية للمشاريع الإنشائية.' : 'High-strength rebar, structural steel beams, cement additives, sealants, and fireproof insulation materials for massive infrastructure construction.',
      icon: iconsMap.buildingMats,
      specs: lang === 'ar'
        ? ['جسور حديد IPE, HEB, HEA مطابقة لمعايير الجودة', 'حديد تسليح عالي المقاومة بمقاسات مختلفة', 'مواد إحكام كيميائية وطلاءات مقاومة للحريق']
        : ['IPE, HEB, HEA structural steel sections', 'High-yield deformed reinforcing rebar arrays', 'Certified fireproofing coatings and sealants'],
      applications: lang === 'ar'
        ? ['المجمعات السكنية والمستودعات الكبرى', 'الجسور ومشاريع البنية التحتية', 'المباني والأبراج الهيكلية']
        : ['Commercial warehouses and logistics hubs', 'Bridges and public transit utilities', 'High-rise structural columns']
    },
    {
      id: 'insulation',
      category: 'building',
      name: lang === 'ar' ? 'مواد العزل الحراري والصوتي والحرائق' : 'Fireproof Insulation Materials',
      desc: lang === 'ar' ? 'ألواح صوف صخري، صوف زجاجي ومواد رغوية مخصصة لعزل الحرارة والحرائق وتوفير استهلاك الطاقة.' : 'High-density rockwool panels, fiberglass wraps, and elastomeric foam tubes for fire protection, acoustics, and thermal efficiency.',
      icon: iconsMap.insulation,
      specs: lang === 'ar'
        ? ['ألواح صوف صخري بكثافة تصل إلى 160 كجم/متر مكعب', 'مقاومة درجات حرارة تصل إلى 1000 درجة مئوية', 'مطابقة لمعايير السلامة وعزل الصوت']
        : ['Rockwool panels up to 160 kg/m³ density', 'Temperature ratings up to 1000°C fire defense', 'Acoustic absorption and noise attenuation limits'],
      applications: lang === 'ar'
        ? ['عزل الغلايات والأنابيب الصناعية الكبرى', 'عزل الجدران والأسقف بالمجمعات والمباني', 'أنظمة التكييف والتهوية المركزية HVAC']
        : ['Refinery boiler and pipe jackets', 'Building envelope wall and partition insulation', 'HVAC duct thermal insulation covers']
    },
    {
      id: 'procure',
      category: 'general',
      name: lang === 'ar' ? 'خدمات التوريد المخصص للشركات' : 'Custom Procurement Solutions',
      desc: lang === 'ar' ? 'تأمين التجهيزات والأدوات الصناعية النادرة والمخصصة وتوريدها مباشرة وفقاً لجدول المواصفات والرسومات الهندسية.' : 'Sourcing and procuring rare, custom-cut components and specialized tools based directly on engineering CAD drawings and BOQs.',
      icon: iconsMap.procure,
      specs: lang === 'ar'
        ? ['فريق مشتريات يبحث عالمياً ومحلياً', 'مراجعة ومطابقة هندسية كاملة للمواصفات', 'توفير شهادات المنشأ والاختبار الكاملة']
        : ['Global sourcing and logistics network access', 'Pre-purchase engineering and specs validation', 'Provision of full origin and certificates documentation'],
      applications: lang === 'ar'
        ? ['مشاريع إحلال وتحديث المصانع والمصافي', 'عقود التوريد السنوية وعقود المشتريات الموحدة', 'تجهيز المواقع الإنشائية الكبرى مسبقاً']
        : ['Refinery modernization and refits', 'Annual framework procurement agreements', 'Mega project staging and preparation supplies']
    },
    {
      id: 'consolidated',
      category: 'general',
      name: lang === 'ar' ? 'الطلبيات اللوجستية المجمعة' : 'Consolidated Industrial Shipments',
      desc: lang === 'ar' ? 'خدمة تجميع وتوحيد التوريدات من مصادر متعددة وتغليفها ونقلها مباشرة إلى موقع العميل بالدمام وأنحاء المملكة.' : 'Consolidating and staging multiple sourcing streams, verifying packaging, and managing direct shipping to worksites across Saudi Arabia.',
      icon: iconsMap.consolidated,
      specs: lang === 'ar'
        ? ['مستودعات مجهزة للتجميع بالدمام', 'فحص وتوثيق المواد قبل التسليم للموقع', 'تقارير متابعة لحظية لحالة سلاسل التوريد']
        : ['50,000+ sq. ft. local warehouse consolidation nodes', 'Quality inspect and packaging review prior to site delivery', 'Real-time supply chain updates and shipping tracking'],
      applications: lang === 'ar'
        ? ['تجهيزات وعقود الصيانة الكبرى (MRO)', 'طلبيات المقاولين الإنشائيين الكبرى المجمعة', 'عقود الإمداد الحكومية واسعة النطاق']
        : ['Large scale turnaround maintenance blocks (MRO)', 'EPC contractor bulk construction setups', 'Government municipal development frameworks']
    }
  ];

  // Mouse move handler for card hover glow
  const handleCardMouseMove = (e: React.MouseEvent<HTMLDivElement>) => {
    const card = e.currentTarget;
    const rect = card.getBoundingClientRect();
    const x = e.clientX - rect.left;
    const y = e.clientY - rect.top;
    card.style.setProperty('--mouse-x', `${x}px`);
    card.style.setProperty('--mouse-y', `${y}px`);
  };

  // Filter items based on active tab selection and search query
  const filteredProducts = productsList.filter(p => {
    const matchesCategory = filter === 'all' || p.category === filter;
    const matchesSearch = searchQuery 
      ? p.name.toLowerCase().includes(searchQuery.toLowerCase()) || 
        p.desc.toLowerCase().includes(searchQuery.toLowerCase())
      : true;
    return matchesCategory && matchesSearch;
  });

  return (
    <section id="products" className="py-24 bg-canvas-soft border-t border-glass-border select-none">
      <div className="max-w-7xl mx-auto px-6">
        
        {/* Section Header */}
        <div className="flex flex-col items-center text-center mb-12">
          <span className="caption-mono text-accent-blue mb-3 block">
            {t.badge}
          </span>
          <h2 className="display-lg text-ink">
            {t.headline}
          </h2>
        </div>

        {/* 7 categories filter tabs */}
        <div className="flex justify-center mb-10 overflow-x-auto pb-4">
          <div className="p-1 glass-panel rounded-full flex gap-1 bg-glass-border/10 whitespace-nowrap min-w-max">
            <button
              onClick={() => setFilter('all')}
              className={`px-4 py-2 rounded-full text-[10px] font-bold uppercase tracking-wider transition-all focus:outline-none cursor-pointer ${
                filter === 'all'
                  ? 'bg-gradient-to-r from-accent-blue to-accent-cyan text-white shadow-sm'
                  : 'text-body hover:text-ink'
              }`}
            >
              {lang === 'ar' ? 'الكل' : 'All Showroom'}
            </button>
            {(['industrial', 'mechanical', 'electrical', 'safety', 'welding', 'building', 'general'] as const).map((cat) => (
              <button
                key={cat}
                onClick={() => setFilter(cat)}
                className={`px-4 py-2 rounded-full text-[10px] font-bold uppercase tracking-wider transition-all focus:outline-none cursor-pointer ${
                  filter === cat
                    ? 'bg-gradient-to-r from-accent-blue to-accent-cyan text-white shadow-sm'
                    : 'text-body hover:text-ink'
                }`}
              >
                {cats[cat]}
              </button>
            ))}
          </div>
        </div>

        {/* Active Search indicators */}
        {searchQuery && (
          <div className="mb-6 text-center text-xs font-mono text-body flex items-center justify-center gap-2">
            <span>
              {lang === 'ar' ? `نتائج البحث عن: "${searchQuery}"` : `Showing search results for: "${searchQuery}"`}
            </span>
            <button 
              onClick={() => setSearchQuery('')}
              className="text-accent-blue underline hover:text-accent-cyan"
            >
              {lang === 'ar' ? 'مسح البحث' : 'Clear Search'}
            </button>
          </div>
        )}

        {/* Product Catalogue Grid */}
        <motion.div 
          layout
          className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4"
        >
          <AnimatePresence>
            {filteredProducts.map((prod, index) => (
              <motion.div
                layout
                key={prod.id}
                initial={{ opacity: 0, scale: 0.95 }}
                animate={{ opacity: 1, scale: 1 }}
                exit={{ opacity: 0, scale: 0.95 }}
                transition={{ duration: 0.4, delay: index * 0.01 }}
                onMouseMove={handleCardMouseMove}
                className="glow-card p-5 glass-panel rounded-xl flex flex-col justify-between h-[235px] group cursor-default"
                id={`products-${prod.id}`}
              >
                <div>
                  {/* Card top */}
                  <div className="flex justify-between items-start mb-3">
                    <div className="w-10 h-10 rounded-lg bg-glass-border/40 flex items-center justify-center border border-glass-border shadow-sm group-hover:scale-105 transition-transform">
                      {prod.icon}
                    </div>
                    <span className="text-[9px] font-mono text-mute uppercase tracking-widest">
                      {cats[prod.category] || prod.category}
                    </span>
                  </div>
                  {/* Title & Short description */}
                  <h3 className="text-xs font-bold text-ink mb-1 group-hover:text-accent-cyan transition-colors">
                    {prod.name}
                  </h3>
                  <p className="text-[10px] text-body leading-relaxed line-clamp-3">
                    {prod.desc}
                  </p>
                </div>

                {/* Card footer CTA buttons */}
                <div className="mt-4 flex justify-between items-center z-10">
                  <button 
                    onClick={() => setSelectedProduct(prod)}
                    className="text-[9px] font-extrabold uppercase tracking-widest text-accent-blue hover:text-accent-cyan transition-colors flex items-center gap-1 focus:outline-none cursor-pointer"
                  >
                    <Eye className="w-3.5 h-3.5" />
                    {t.viewDetails}
                  </button>
                  <button 
                    onClick={onRequestQuote}
                    className="px-3 py-1.5 rounded-full border border-glass-border bg-canvas/30 hover:bg-glass-border/50 text-[9px] font-extrabold uppercase tracking-wider text-ink transition-colors cursor-pointer"
                  >
                    {t.contactSales}
                  </button>
                </div>
              </motion.div>
            ))}
          </AnimatePresence>
        </motion.div>

        {/* Empty catalog indicator */}
        {filteredProducts.length === 0 && (
          <div className="p-12 glass-panel rounded-2xl text-center max-w-md mx-auto">
            <p className="text-body font-mono text-xs">
              {lang === 'ar' ? 'عذرًا، لم يتم العثور على أي منتج يطابق معايير البحث.' : 'No products match your search query.'}
            </p>
          </div>
        )}

        {/* Technical Specification Modal */}
        <AnimatePresence>
          {selectedProduct && (
            <div className="fixed inset-0 z-50 flex items-center justify-center p-4">
              {/* Backdrop */}
              <motion.div
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                exit={{ opacity: 0 }}
                onClick={() => setSelectedProduct(null)}
                className="absolute inset-0 bg-[#050816]/70 backdrop-blur-sm"
              />

              {/* Modal window */}
              <motion.div
                initial={{ opacity: 0, scale: 0.9, y: 30 }}
                animate={{ opacity: 1, scale: 1, y: 0 }}
                exit={{ opacity: 0, scale: 0.9, y: 20 }}
                transition={{ duration: 0.3 }}
                className="relative max-w-md w-full glass-panel p-6 rounded-2xl shadow-2xl z-10 text-ink border-t border-t-accent-cyan/35 max-h-[85vh] overflow-y-auto"
              >
                {/* Close */}
                <button
                  onClick={() => setSelectedProduct(null)}
                  className="absolute top-4 right-4 p-1.5 rounded-full hover:bg-glass-border/30 text-body hover:text-ink transition-colors focus:outline-none"
                  aria-label="Close modal"
                >
                  <X className="w-5 h-5" />
                </button>

                {/* Header info */}
                <div className="flex items-center gap-3 mb-5 mt-2 text-left rtl:text-right">
                  <div className="w-12 h-12 rounded-xl bg-glass-border/40 flex items-center justify-center border border-glass-border shadow-sm flex-shrink-0">
                    {selectedProduct.icon}
                  </div>
                  <div>
                    <span className="text-[9px] font-mono text-mute uppercase tracking-widest block mb-0.5">
                      {cats[selectedProduct.category]} range
                    </span>
                    <h4 className="text-sm font-bold leading-tight">{selectedProduct.name}</h4>
                  </div>
                </div>

                {/* Detailed descriptions */}
                <p className="text-xs text-body leading-relaxed mb-6 text-left rtl:text-right">
                  {selectedProduct.desc}
                </p>

                {/* Technical Specs list */}
                <div className="text-left rtl:text-right mb-5">
                  <h5 className="text-[9px] font-bold uppercase tracking-wider text-mute mb-2.5 font-mono">
                    {lang === 'ar' ? 'المواصفات الفنية المعتمدة' : 'Technical Specifications'}
                  </h5>
                  <ul className="space-y-2">
                    {selectedProduct.specs.map((spec, i) => (
                      <li key={i} className="flex items-start gap-2 text-xs">
                        <div className="w-1.5 h-1.5 rounded-full bg-accent-cyan mt-1.5 flex-shrink-0" />
                        <span className="text-body leading-tight">{spec}</span>
                      </li>
                    ))}
                  </ul>
                </div>

                {/* Applications list */}
                <div className="text-left rtl:text-right mb-6">
                  <h5 className="text-[9px] font-bold uppercase tracking-wider text-mute mb-2.5 font-mono">
                    {lang === 'ar' ? 'مجالات وتطبيقات الاستخدام' : 'Industrial Applications'}
                  </h5>
                  <ul className="space-y-2">
                    {selectedProduct.applications.map((app, i) => (
                      <li key={i} className="flex items-start gap-2 text-xs">
                        <div className="w-1.5 h-1.5 rounded-full bg-accent-blue mt-1.5 flex-shrink-0" />
                        <span className="text-body leading-tight">{app}</span>
                      </li>
                    ))}
                  </ul>
                </div>

                {/* Conversion triggers: WhatsApp & RFQ Quote */}
                <div className="flex flex-col sm:flex-row items-center gap-3">
                  <a
                    href={`https://wa.me/966563539227?text=${encodeURIComponent(
                      lang === 'ar' 
                        ? `مرحباً مؤسسة أوبتيموم للتوريد التجاري،\nأود الاستفسار عن كفاءة وأسعار المنتج: ${selectedProduct.name}.\nالرجاء تزويدي بالمواصفات الفنية ومدى توفرها.`
                        : `Hello Optimum Supply Trading Est.,\nI am interested in your: ${selectedProduct.name}.\nPlease send me pricing, technical specifications, and delivery availability. Thank you.`
                    )}`}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="w-full sm:flex-1 px-4 py-2.5 rounded-xl bg-gradient-to-r from-emerald-500 to-green-600 hover:opacity-95 text-white text-xs font-semibold text-center flex items-center justify-center gap-1.5 shadow-sm"
                  >
                    <Send className="w-4 h-4" />
                    {t.chatWhatsapp}
                  </a>
                  
                  <button
                    onClick={() => {
                      setSelectedProduct(null);
                      onRequestQuote();
                    }}
                    className="w-full sm:w-auto px-4 py-2.5 rounded-xl border border-glass-border hover:bg-glass-border/30 text-ink text-xs font-semibold flex items-center justify-center gap-1.5 cursor-pointer"
                  >
                    <Mail className="w-4 h-4 text-accent-blue" />
                    {lang === 'ar' ? 'طلب تسعيرة' : 'Request Quote'}
                  </button>
                </div>
              </motion.div>
            </div>
          )}
        </AnimatePresence>
      </div>
    </section>
  );
};
export default ProductShowcase;
