// portfolio/src/components/ThreeDHeroCanvas.tsx
import React, { useRef, useMemo, useState, useEffect } from 'react';
import { Canvas, useFrame, useThree } from '@react-three/fiber';
import { Line } from '@react-three/drei';
import * as THREE from 'three';

// ─── CAMERA CONTROLLER (Scroll & Mouse Parallax) ───
const CameraController: React.FC = () => {
  const { camera } = useThree();
  const [scrollY, setScrollY] = useState(0);
  const mouse = useRef({ x: 0, y: 0 });

  useEffect(() => {
    const handleScroll = () => {
      setScrollY(window.scrollY);
    };

    const handleMouseMove = (e: MouseEvent) => {
      // Normalize to -1 to 1 range
      mouse.current.x = (e.clientX / window.innerWidth) * 2 - 1;
      mouse.current.y = -(e.clientY / window.innerHeight) * 2 + 1;
    };

    window.addEventListener('scroll', handleScroll, { passive: true });
    window.addEventListener('mousemove', handleMouseMove);
    return () => {
      window.removeEventListener('scroll', handleScroll);
      window.removeEventListener('mousemove', handleMouseMove);
    };
  }, []);

  useFrame(({ clock }) => {
    const time = clock.getElapsedTime();
    
    // Circular orbit radius decreases slightly as user scrolls (zoom in effect)
    const baseRadius = 5.2 - Math.min(2.5, scrollY * 0.0035);
    // Slow rotational baseline + mouse panning offset
    const angle = time * 0.045 + (mouse.current.x * 0.18);
    
    const targetX = Math.sin(angle) * baseRadius;
    const targetZ = Math.cos(angle) * baseRadius;
    // Panning height adjusts dynamically with scroll and mouse movement
    const targetY = 1.0 + Math.sin(time * 0.12) * 0.3 + (mouse.current.y * 0.3) - (scrollY * 0.0008);

    // Smoothly interpolate camera positioning
    camera.position.x += (targetX - camera.position.x) * 0.06;
    camera.position.y += (targetY - camera.position.y) * 0.06;
    camera.position.z += (targetZ - camera.position.z) * 0.06;

    // Point camera towards center of refinery complex
    camera.lookAt(0, 0.3, 0);
  });

  return null;
};

// ─── COMPONENT: EXHAUST STEAM PARTICLES ───
interface SteamProps {
  position: [number, number, number];
}

const ExhaustSteam: React.FC<SteamProps> = ({ position }) => {
  const pointsRef = useRef<THREE.Points>(null);
  const count = 30;

  const [positions, speeds, scales] = useMemo(() => {
    const pos = new Float32Array(count * 3);
    const sp = new Float32Array(count);
    const sc = new Float32Array(count);
    for (let i = 0; i < count; i++) {
      // Spawn at chimney top with small random horizontal deviation
      pos[i * 3] = position[0] + (Math.random() - 0.5) * 0.1;
      pos[i * 3 + 1] = position[1] + Math.random() * 1.5;
      pos[i * 3 + 2] = position[2] + (Math.random() - 0.5) * 0.1;

      sp[i] = 0.4 + Math.random() * 0.5; // rising velocity
      sc[i] = 0.02 + Math.random() * 0.05; // size
    }
    return [pos, sp, sc];
  }, [position]);

  useFrame((state, delta) => {
    if (pointsRef.current) {
      const posArr = pointsRef.current.geometry.attributes.position.array as Float32Array;
      const time = state.clock.getElapsedTime();

      for (let i = 0; i < count; i++) {
        // Rise up
        posArr[i * 3 + 1] += speeds[i] * delta;
        // Float side to side like smoke in breeze
        posArr[i * 3] += Math.sin(time * 2 + i) * 0.003;

        // Reset if float height goes beyond threshold
        if (posArr[i * 3 + 1] > position[1] + 1.8) {
          posArr[i * 3 + 1] = position[1];
          posArr[i * 3] = position[0] + (Math.random() - 0.5) * 0.08;
          posArr[i * 3 + 2] = position[2] + (Math.random() - 0.5) * 0.08;
        }
      }
      pointsRef.current.geometry.attributes.position.needsUpdate = true;
    }
  });

  return (
    <points ref={pointsRef}>
      <bufferGeometry>
        <bufferAttribute
          attach="attributes-position"
          args={[positions, 3]}
        />
      </bufferGeometry>
      <pointsMaterial
        size={0.12}
        color="#a1a1a1"
        transparent
        opacity={0.35}
        depthWrite={false}
      />
    </points>
  );
};

// ─── COMPONENT: ROTATING MACHINE GEARS ───
interface GearProps {
  position: [number, number, number];
  radius: number;
  teeth: number;
  rotationSpeed?: number;
  axis?: 'x' | 'y' | 'z';
}

const RotatingIndustrialGear: React.FC<GearProps> = ({ 
  position, radius, teeth, rotationSpeed = 0.4, axis = 'z' 
}) => {
  const groupRef = useRef<THREE.Group>(null);

  useFrame((state) => {
    if (groupRef.current) {
      const angle = state.clock.getElapsedTime() * rotationSpeed;
      if (axis === 'z') groupRef.current.rotation.z = angle;
      else if (axis === 'x') groupRef.current.rotation.x = angle;
      else groupRef.current.rotation.y = angle;
    }
  });

  // Calculate teeth segments
  const teethMeshes = useMemo(() => {
    const list = [];
    const toothWidth = (2 * Math.PI * radius) / (teeth * 2.2);
    for (let i = 0; i < teeth; i++) {
      const theta = (i / teeth) * Math.PI * 2;
      const x = Math.sin(theta) * radius;
      const y = Math.cos(theta) * radius;
      list.push(
        <mesh 
          key={i} 
          position={[x, y, 0]} 
          rotation={[0, 0, -theta]}
        >
          <boxGeometry args={[toothWidth, 0.15, 0.15]} />
          <meshStandardMaterial color="#1e293b" metalness={0.9} roughness={0.1} />
        </mesh>
      );
    }
    return list;
  }, [radius, teeth]);

  return (
    <group position={position}>
      <group ref={groupRef}>
        {/* Core center cylinder */}
        <mesh rotation={[Math.PI / 2, 0, 0]}>
          <cylinderGeometry args={[radius * 0.9, radius * 0.9, 0.12, 16]} />
          <meshStandardMaterial color="#1e293b" metalness={0.9} roughness={0.15} />
        </mesh>
        
        {/* Center hub */}
        <mesh rotation={[Math.PI / 2, 0, 0]}>
          <cylinderGeometry args={[radius * 0.25, radius * 0.25, 0.16, 8]} />
          <meshStandardMaterial color="#0066ff" metalness={0.95} roughness={0.05} />
        </mesh>
        
        {/* Teeth */}
        {teethMeshes}
      </group>
    </group>
  );
};

// ─── COMPONENT: ANIMATED CONSTRUCTION JIB CRANE ───
const AnimatedIndustrialCrane: React.FC = () => {
  const jibRef = useRef<THREE.Group>(null);

  useFrame((state) => {
    if (jibRef.current) {
      // Slow back and forth sweep rotation
      jibRef.current.rotation.y = Math.sin(state.clock.getElapsedTime() * 0.15) * 0.6;
    }
  });

  return (
    <group position={[-2.3, -1.8, -1.5]}>
      {/* Crane Base Tower (Lattice Truss structure) */}
      <mesh position={[0, 1.2, 0]}>
        <cylinderGeometry args={[0.08, 0.12, 2.4, 4, 1, true]} />
        <meshStandardMaterial color="#1e293b" wireframe metalness={0.8} roughness={0.2} />
      </mesh>

      {/* Rotating Header and Jib Arm */}
      <group ref={jibRef} position={[0, 2.4, 0]}>
        {/* Rotating Cab / Counterweight */}
        <mesh position={[0, 0.1, 0]}>
          <boxGeometry args={[0.2, 0.2, 0.25]} />
          <meshStandardMaterial color="#ff8c00" metalness={0.8} roughness={0.15} />
        </mesh>
        
        {/* Long Horizontal Jib arm */}
        <mesh position={[0.6, 0.1, 0]} rotation={[0, 0, 0]}>
          <boxGeometry args={[1.4, 0.05, 0.05]} />
          <meshStandardMaterial color="#ff8c00" wireframe />
        </mesh>
        
        {/* Counter Jib weight */}
        <mesh position={[-0.3, 0.1, 0]}>
          <boxGeometry args={[0.5, 0.08, 0.08]} />
          <meshStandardMaterial color="#334155" metalness={0.9} />
        </mesh>

        {/* Hanging hoist wire */}
        <Line 
          points={[[1.1, 0.1, 0], [1.1, -1.0, 0]]}
          color="#a1a1a1"
          lineWidth={1.0}
        />
        {/* Hook block */}
        <mesh position={[1.1, -1.0, 0]}>
          <sphereGeometry args={[0.04, 8, 8]} />
          <meshStandardMaterial color="#ff8c00" metalness={0.9} />
        </mesh>
      </group>
    </group>
  );
};

// ─── COMPONENT: REFINERY PIPELINES ───
const ShinyPipes: React.FC = () => {
  return (
    <group>
      {/* Horizontal Connector Pipe */}
      <mesh position={[-0.1, 0.8, -0.6]} rotation={[0, 0, Math.PI / 2]}>
        <cylinderGeometry args={[0.06, 0.06, 2.2, 8]} />
        <meshStandardMaterial color="#475569" metalness={0.98} roughness={0.05} />
      </mesh>
      {/* Right Elbow joints */}
      <mesh position={[1.0, 0.8, -0.6]}>
        <sphereGeometry args={[0.07, 8, 8]} />
        <meshStandardMaterial color="#ff8c00" metalness={0.95} />
      </mesh>
      {/* Down pipe */}
      <mesh position={[1.0, 0.1, -0.6]}>
        <cylinderGeometry args={[0.06, 0.06, 1.4, 8]} />
        <meshStandardMaterial color="#475569" metalness={0.98} roughness={0.05} />
      </mesh>

      {/* Elevated pipeline running across floor */}
      <mesh position={[0, -1.75, 0.5]} rotation={[Math.PI / 2, 0, 0]}>
        <cylinderGeometry args={[0.05, 0.05, 4.0, 8]} />
        <meshStandardMaterial color="#334155" metalness={0.95} roughness={0.1} />
      </mesh>
      <mesh position={[0, -1.75, -0.5]} rotation={[Math.PI / 2, 0, 0]}>
        <cylinderGeometry args={[0.05, 0.05, 4.0, 8]} />
        <meshStandardMaterial color="#334155" metalness={0.95} roughness={0.1} />
      </mesh>
    </group>
  );
};

// ─── COMPONENT: REFINERY STRUCTURES (Towers, Tanks) ───
const RefineryStructures: React.FC = () => {
  return (
    <group>
      {/* Main Fractionating Column Tower */}
      <group position={[-1.2, -0.8, -0.6]}>
        <mesh>
          <cylinderGeometry args={[0.36, 0.36, 3.4, 16]} />
          <meshStandardMaterial 
            color="#334155" 
            metalness={0.95} 
            roughness={0.08} 
          />
        </mesh>
        
        {/* Structural Rings */}
        <mesh position={[0, 1.0, 0]} rotation={[Math.PI / 2, 0, 0]}>
          <torusGeometry args={[0.39, 0.03, 8, 24]} />
          <meshStandardMaterial color="#ff8c00" metalness={0.9} />
        </mesh>
        <mesh position={[0, 0.2, 0]} rotation={[Math.PI / 2, 0, 0]}>
          <torusGeometry args={[0.39, 0.03, 8, 24]} />
          <meshStandardMaterial color="#1e293b" metalness={0.9} />
        </mesh>
        <mesh position={[0, -0.6, 0]} rotation={[Math.PI / 2, 0, 0]}>
          <torusGeometry args={[0.39, 0.03, 8, 24]} />
          <meshStandardMaterial color="#ff8c00" metalness={0.9} />
        </mesh>

        {/* Chimney Tip Flare */}
        <mesh position={[0, 1.7, 0]}>
          <cylinderGeometry args={[0.18, 0.28, 0.4, 12]} />
          <meshStandardMaterial color="#1e293b" metalness={0.9} />
        </mesh>
        
        {/* Flare Indicator Point Light */}
        <pointLight position={[0, 2.0, 0]} intensity={4.5} color="#ff8c00" distance={2.5} decay={2} />
      </group>

      {/* Steam Smoke Emitters on top of Chimney */}
      <ExhaustSteam position={[-1.2, 0.9, -0.6]} />

      {/* Secondary Refinery Stack (Thinner) */}
      <group position={[-0.8, -0.6, -1.2]}>
        <mesh>
          <cylinderGeometry args={[0.15, 0.15, 3.8, 12]} />
          <meshStandardMaterial color="#1e293b" metalness={0.95} roughness={0.05} />
        </mesh>
        <pointLight position={[0, 2.0, 0]} intensity={2.0} color="#ff3300" distance={1.8} />
      </group>
      <ExhaustSteam position={[-0.8, 1.3, -1.2]} />

      {/* Big Storage Tank */}
      <group position={[1.2, -1.2, -0.2]}>
        <mesh>
          <cylinderGeometry args={[1.0, 1.0, 1.5, 24]} />
          <meshStandardMaterial color="#334155" metalness={0.9} roughness={0.12} />
        </mesh>
        
        {/* Support dome cap */}
        <mesh position={[0, 0.75, 0]}>
          <sphereGeometry args={[1.0, 24, 12, 0, Math.PI * 2, 0, Math.PI / 2]} />
          <meshStandardMaterial color="#ff8c00" metalness={0.95} roughness={0.08} />
        </mesh>

        {/* Pulsing Safety Light on Dome */}
        <pointLight position={[0, 1.2, 0]} intensity={3} color="#0066ff" distance={2.5} />
      </group>

      {/* Medium Chemical Storage Tank */}
      <group position={[0.2, -1.3, -1.4]}>
        <mesh>
          <cylinderGeometry args={[0.7, 0.7, 1.3, 16]} />
          <meshStandardMaterial color="#475569" metalness={0.88} roughness={0.15} />
        </mesh>
        <mesh position={[0, 0.65, 0]}>
          <sphereGeometry args={[0.7, 16, 8, 0, Math.PI * 2, 0, Math.PI / 2]} />
          <meshStandardMaterial color="#334155" metalness={0.9} />
        </mesh>
      </group>
    </group>
  );
};

// ─── COMPONENT: FLOOR GRID PLATFORM ───
const FloorGrid: React.FC = () => {
  return (
    <group position={[0, -1.95, 0]}>
      {/* Industrial Grid Base Helper */}
      <gridHelper
        args={[16, 16, '#0066ff', '#1e293b']}
        position={[0, 0, 0]}
      />
      {/* Steel structure floor frame */}
      <mesh position={[0, -0.05, 0]}>
        <boxGeometry args={[16, 0.05, 16]} />
        <meshStandardMaterial color="#050816" opacity={0.6} transparent roughness={0.9} />
      </mesh>
    </group>
  );
};

// ─── COMPONENT: AMBIENT GLOW DUST PARTICLES ───
const GlowDust: React.FC = () => {
  const count = 70;
  const pointsRef = useRef<THREE.Points>(null);

  const [positions, speeds] = useMemo(() => {
    const pos = new Float32Array(count * 3);
    const sp = new Float32Array(count);
    for (let i = 0; i < count; i++) {
      pos[i * 3] = (Math.random() - 0.5) * 8; // X
      pos[i * 3 + 1] = -1.8 + Math.random() * 4; // Y
      pos[i * 3 + 2] = (Math.random() - 0.5) * 8; // Z
      sp[i] = 0.04 + Math.random() * 0.08;
    }
    return [pos, sp];
  }, []);

  useFrame((state, delta) => {
    if (pointsRef.current) {
      const pos = pointsRef.current.geometry.attributes.position.array as Float32Array;
      const time = state.clock.getElapsedTime();

      for (let i = 0; i < count; i++) {
        // Slow vertical drift
        pos[i * 3 + 1] += speeds[i] * delta;
        // Wavy drift
        pos[i * 3] += Math.sin(time * 0.5 + i) * 0.001;

        if (pos[i * 3 + 1] > 3.0) {
          pos[i * 3 + 1] = -1.8;
          pos[i * 3] = (Math.random() - 0.5) * 8;
        }
      }
      pointsRef.current.geometry.attributes.position.needsUpdate = true;
    }
  });

  return (
    <points ref={pointsRef}>
      <bufferGeometry>
        <bufferAttribute
          attach="attributes-position"
          args={[positions, 3]}
        />
      </bufferGeometry>
      <pointsMaterial
        size={0.04}
        color="#0066ff"
        transparent
        opacity={0.6}
        depthWrite={false}
      />
    </points>
  );
};

// ─── MAIN CANVAS CONTAINER ───
export const ThreeDHeroCanvas: React.FC = () => {
  return (
    <div className="w-full h-full">
      <Canvas
        camera={{ position: [0, 0.8, 5.0], fov: 50 }}
        gl={{ antialias: true, alpha: true }}
      >
        {/* Soft atmospheric base lights */}
        <ambientLight intensity={1.2} />
        <directionalLight position={[5, 10, 3]} intensity={2.0} color="#ffffff" />
        
        {/* Volumetric / Spot light beams mapping from high stack */}
        <spotLight 
          position={[-1.2, 1.8, -0.6]} 
          angle={Math.PI / 6} 
          penumbra={0.6} 
          intensity={6} 
          color="#0066ff" 
          distance={8} 
          decay={1.5}
        />
        <spotLight 
          position={[1.2, 1.2, -0.2]} 
          angle={Math.PI / 5} 
          penumbra={0.7} 
          intensity={5} 
          color="#ff8c00" 
          distance={6} 
          decay={1.5}
        />

        {/* Fog simulation for depth */}
        <fog attach="fog" args={['#050816', 3.0, 10.0]} />

        {/* 3D Refinery Meshes */}
        <FloorGrid />
        <RefineryStructures />
        <ShinyPipes />
        
        {/* Rotating Mechanical Parts */}
        <RotatingIndustrialGear position={[1.2, -0.85, 0.85]} radius={0.35} teeth={10} rotationSpeed={0.5} />
        <RotatingIndustrialGear position={[1.65, -0.85, 0.85]} radius={0.2} teeth={6} rotationSpeed={-0.8} />
        
        {/* Cranes */}
        <AnimatedIndustrialCrane />

        {/* Particle systems */}
        <GlowDust />

        {/* Cinematic Scroll & Mouse pointer controls */}
        <CameraController />
      </Canvas>
    </div>
  );
};
export default ThreeDHeroCanvas;
