// portfolio/src/components/ThreeDHero.tsx
import React, { useState, useEffect } from 'react';
import { motion } from 'framer-motion';
import { ArrowRight, Shield, Award, Users, CheckCircle2, MessageSquare, Package, Clock } from 'lucide-react';
import { translations } from '../data/translations';
import ThreeDHeroCanvas from './ThreeDHeroCanvas';

interface ThreeDHeroProps {
  currentLang: 'en' | 'ar';
  onRequestQuote: () => void;
}

export const ThreeDHero: React.FC<ThreeDHeroProps> = ({ 
  currentLang: initialLang,
  onRequestQuote
}) => {
  const [lang, setLang] = useState<'en' | 'ar'>(initialLang);

  // Sync language selection
  useEffect(() => {
    const handleLangChange = (e: Event) => {
      const customEvent = e as CustomEvent<'en' | 'ar'>;
      setLang(customEvent.detail);
    };

    window.addEventListener('lang-change', handleLangChange);
    return () => window.removeEventListener('lang-change', handleLangChange);
  }, []);

  const t = translations[lang].hero;

  const containerVariants = {
    hidden: { opacity: 0 },
    visible: {
      opacity: 1,
      transition: {
        staggerChildren: 0.15,
        delayChildren: 0.2,
      },
    },
  };

  const itemVariants = {
    hidden: { y: 25, opacity: 0 },
    visible: {
      y: 0,
      opacity: 1,
      transition: { duration: 0.8, ease: [0.16, 1, 0.3, 1] },
    },
  };

  return (
    <section className="relative w-full min-h-screen flex items-center justify-center overflow-hidden mesh-gradient pt-16 select-none">
      {/* 3D Canvas Background Container */}
      <div className="absolute inset-0 z-0 opacity-85 dark:opacity-100">
        <ThreeDHeroCanvas />
      </div>

      {/* Radial shade covers for atmospheric lighting */}
      <div className="absolute top-0 left-0 right-0 h-32 bg-gradient-to-b from-canvas-soft to-transparent pointer-events-none z-1" />
      <div className="absolute bottom-0 left-0 right-0 h-40 bg-gradient-to-t from-canvas-soft to-transparent pointer-events-none z-1" />

      {/* Main Grid Content Container */}
      <div className="relative max-w-7xl mx-auto px-6 grid grid-cols-1 lg:grid-cols-12 gap-8 z-10 w-full pt-12 pb-20 items-center">
        {/* Left Column: Rebranded Headlines and Call Actions */}
        <motion.div
          variants={containerVariants}
          initial="hidden"
          animate="visible"
          className="lg:col-span-7 flex flex-col items-start text-left rtl:items-start rtl:text-right"
        >
          {/* Engineering Badge */}
          <motion.div
            variants={itemVariants}
            className="inline-flex items-center gap-2 px-3.5 py-1.5 rounded-full border border-accent-blue/30 bg-accent-blue/10 dark:bg-accent-blue/5 text-accent-blue dark:text-accent-cyan text-xs font-mono font-bold uppercase tracking-wider mb-6 cursor-pointer hover:border-accent-cyan/50 hover:bg-accent-blue/10 transition-colors shadow-sm"
          >
            <span className="relative flex h-2 w-2">
              <span className="animate-ping absolute inline-flex h-full w-full rounded-full bg-accent-cyan opacity-75"></span>
              <span className="relative inline-flex rounded-full h-2 w-2 bg-accent-cyan"></span>
            </span>
            {t.badge}
          </motion.div>

          {/* Top Rating Star Group */}
          <motion.div
            variants={itemVariants}
            className="flex items-center gap-2 mb-6 select-none"
          >
            <div className="flex items-center gap-0.5">
              {[1, 2, 3, 4].map((i) => (
                <svg key={i} className="w-4.5 h-4.5 text-amber-400 fill-amber-400 drop-shadow-[0_0_8px_rgba(251,191,36,0.4)]" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
                  <path d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" />
                </svg>
              ))}
              <svg className="w-4.5 h-4.5 text-amber-400" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
                <defs>
                  <linearGradient id="star-half-grad">
                    <stop offset="50%" stopColor="#fbbf24" />
                    <stop offset="50%" stopColor="#4b5563" stopOpacity="0.4" />
                  </linearGradient>
                </defs>
                <path fill="url(#star-half-grad)" d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" />
              </svg>
            </div>
            <span className="text-[11px] font-extrabold font-sans text-ink">
              {lang === 'ar' ? (
                <span>٤.٥/٥ (أفضل مراجعة من ٢٠,٠٠٠+ مقاول ومهندس)</span>
              ) : (
                <span>4.5/5 (20,000+ Best Reviews from Engineers & Contractors)</span>
              )}
            </span>
          </motion.div>

          {/* Headline */}
          <motion.h1
            variants={itemVariants}
            className="text-4xl sm:text-5xl md:text-6xl font-extrabold text-ink font-sans tracking-tight leading-none mb-6 relative"
          >
            {lang === 'ar' ? (
              <>
                شريك التوريد <br />
                <span className="text-transparent bg-clip-text bg-gradient-to-r from-accent-blue to-accent-cyan drop-shadow-[0_2px_15px_rgba(6,182,212,0.15)]">الموثوق</span> بمشروعك
              </>
            ) : (
              <>
                Your Trusted <br />
                Supply & Sourcing <br />
                <span className="text-transparent bg-clip-text bg-gradient-to-r from-accent-blue to-accent-cyan drop-shadow-[0_2px_15px_rgba(6,182,212,0.15)]">Partner.</span>
              </>
            )}
          </motion.h1>

          {/* Subheadline description */}
          <motion.p
            variants={itemVariants}
            className="text-base sm:text-lg text-body mb-8 max-w-xl font-medium leading-relaxed"
          >
            {t.subheadline}
          </motion.p>

          {/* Sourcing CTAs */}
          <motion.div
            variants={itemVariants}
            className="flex flex-wrap items-center gap-3 w-full sm:w-auto"
          >
            <button
              onClick={onRequestQuote}
              className="px-6 py-3 rounded-full bg-gradient-to-r from-accent-blue to-accent-cyan text-white font-semibold text-sm hover:opacity-95 hover:shadow-lg hover:shadow-accent-cyan/20 active:scale-98 transition-all flex items-center gap-2 group cursor-pointer focus:outline-none"
            >
              {t.requestQuoteBtn}
              <ArrowRight className="w-4.5 h-4.5 group-hover:translate-x-1 transition-transform duration-200 rtl:rotate-180" />
            </button>
            <a
              href="https://wa.me/966563539227"
              target="_blank"
              rel="noopener noreferrer"
              className="px-6 py-3 rounded-full bg-canvas text-ink border border-glass-border font-semibold text-sm hover:bg-glass-border/40 hover:text-ink transition-colors active:scale-98 cursor-pointer flex items-center gap-2"
            >
              <MessageSquare className="w-4.5 h-4.5 text-emerald-500" />
              {t.contactBtn}
            </a>
          </motion.div>
        </motion.div>

        {/* Right Column: Glassmorphic statistics grids */}
        <div className="lg:col-span-5 w-full flex flex-col gap-4 self-center lg:pl-4">
          <motion.div
            initial={{ opacity: 0, x: 30 }}
            animate={{ opacity: 1, x: 0 }}
            transition={{ duration: 0.8, delay: 0.5, ease: 'easeOut' }}
            className="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-2 gap-4 w-full"
          >
            {/* Products */}
            <div className="p-5 glass-panel rounded-xl hover:-translate-y-1 transition-transform duration-300 flex flex-col justify-between h-32 select-none group border-l-2 border-l-accent-blue">
              <div className="flex justify-between items-start text-accent-blue">
                <Package className="w-6 h-6 group-hover:scale-110 transition-transform" />
                <span className="text-2xl font-black font-mono">1000+</span>
              </div>
              <span className="text-[10px] font-bold text-body tracking-wider mt-4 uppercase">
                {lang === 'ar' ? 'المنتجات والحلول' : 'Products & Catalog'}
              </span>
            </div>

            {/* Clients */}
            <div className="p-5 glass-panel rounded-xl hover:-translate-y-1 transition-transform duration-300 flex flex-col justify-between h-32 select-none group border-l-2 border-l-accent-cyan">
              <div className="flex justify-between items-start text-accent-cyan">
                <Users className="w-6 h-6 group-hover:scale-110 transition-transform" />
                <span className="text-2xl font-black font-mono">500+</span>
              </div>
              <span className="text-[10px] font-bold text-body tracking-wider mt-4 uppercase">
                {lang === 'ar' ? 'العملاء المعتمدين' : 'Corporate Clients'}
              </span>
            </div>

            {/* Partners */}
            <div className="p-5 glass-panel rounded-xl hover:-translate-y-1 transition-transform duration-300 flex flex-col justify-between h-32 select-none group border-l-2 border-l-purple-500">
              <div className="flex justify-between items-start text-purple-500">
                <Award className="w-6 h-6 group-hover:scale-110 transition-transform" />
                <span className="text-2xl font-black font-mono">50+</span>
              </div>
              <span className="text-[10px] font-bold text-body tracking-wider mt-4 uppercase">
                {lang === 'ar' ? 'الشركاء الصناعيين' : 'Global Partners'}
              </span>
            </div>

            {/* Support */}
            <div className="p-5 glass-panel rounded-xl hover:-translate-y-1 transition-transform duration-300 flex flex-col justify-between h-32 select-none group border-l-2 border-l-amber-500">
              <div className="flex justify-between items-start text-amber-500">
                <Clock className="w-6 h-6 group-hover:scale-110 transition-transform" />
                <span className="text-2xl font-black font-mono">24/7</span>
              </div>
              <span className="text-[10px] font-bold text-body tracking-wider mt-4 uppercase">
                {lang === 'ar' ? 'الدعم الفني والرد' : 'Procurement Support'}
              </span>
            </div>

            {/* Commitment */}
            <div className="p-5 glass-panel rounded-xl hover:-translate-y-1 transition-transform duration-300 flex flex-col justify-between h-32 select-none group border-l-2 border-l-green-500 col-span-2 lg:col-span-2">
              <div className="flex justify-between items-start text-green-500">
                <CheckCircle2 className="w-6 h-6 group-hover:scale-110 transition-transform" />
                <span className="text-2xl font-black font-mono">100%</span>
              </div>
              <span className="text-[10px] font-bold text-body tracking-wider mt-4 uppercase">
                {lang === 'ar' ? 'الالتزام بالجودة والنزاهة' : 'Quality Commitment'}
              </span>
            </div>
          </motion.div>
        </div>
      </div>
    </section>
  );
};
export default ThreeDHero;
